src/main/java/com/conveyal/r5/streets/StreetRouter.java

```diff
@@ -228,8 +228,8 @@ public boolean setOrigin (double lat, double lon) {
         queue.clear();
         // from vertex is at end of back edge. Set edge correctly so that turn restrictions/costs are applied correctly
         // at the origin.
-        State startState0 = new State(split.vertex0, split.edge + 1, streetMode);
-        State startState1 = new State(split.vertex1, split.edge, streetMode);
+        State startState0 = new State(split.vertex0, -1, streetMode);
+        State startState1 = new State(split.vertex1, -1, streetMode);
```

Comment:
Problem was that when states were returned from StreetRouter they had negative edges ID with ID different from -1. And this broke all of getEdge function since negative edges weren't found in EdgeStore. But If I changed StreetSegment class (which reads states and gets edge geometry to create turn instruction) to skip all negative edges I get path with holes some edges get skipped.

I'm not sure if edges with negative IDs are even supposed to be returned from StreetRouter.

So the only solution that worked for both P+R and bike share was this: https://github.com/conveyal/r5/commit/c0907daae49ffb32c68d96381018dc64fbac31d4